### PR TITLE
Add 2.4-beta1 release notes

### LINKED ISSUE #2642: Add 2.4-beta1 release notes

### PR DESCRIPTION
**Pull request by pmatulis**
resolves #2642

### CODE REVIEW

File: src/en/reference-release-notes.md [hunk 1]
```diff
@@ -15,6 +15,167 @@ CARRIAGE RETURN; LISTS ARE TOO HEAVY-HANDED IMO
...
+  There is also a known issue that trying to `juju remove-machine` the machine
+  that is currently the Mongo primary will not cleanup properly. This should be
+  addressed in the next 2.4 release.
+  
+  In future 2.4 releases, we will also be updating `juju enable-ha` to remove its
+  logic around demoting machines. Instead, `juju enable-ha` will only be a way to
+  ensure that you have the correct number of controller machines being
+  started/intended to participate in HA. This will also fix issues around
+  launching 2 new machines (going to 5) while machines 2 and 3 are still
+  starting.h
```
Comment (evilnick) [inline src/en/reference-release-notes.md]: there's an extra 'h' here

File: src/en/reference-release-notes.md [hunk 2]
```diff
@@ -15,6 +15,167 @@ CARRIAGE RETURN; LISTS ARE TOO HEAVY-HANDED IMO
...
+  specified in the feedback section.
+  
+  ## Get Juju.
+  
+  The only way to get the development version of Juju is by using the `--edge`
+  channel in the `snap` package:
+  
+  	  sudo snap install juju --edge --classic
+
+  To move to the stable version:
```
Comment (evilnick) [inline src/en/reference-release-notes.md]: I would say, "You can revert your Juju client to the latest stable version with: "

Comment (pmatulis) [reply src/en/reference-release-notes.md]: Doesn't "revert" imply going back to what already was?

File: src/en/reference-release-notes.md (hunk 1 above)
Comment (pmatulis) [reply src/en/reference-release-notes.md]: :ok:

File: src/en/reference-release-notes.md (hunk 2 above)
Comment (evilnick) [reply src/en/reference-release-notes.md]: the point of putting that in there is to tell people how to get back to the stable version, i guess assuming that they had a stable version and then used tohe above instruction to try the new beta.
Word it however you like. The current wording isn't clear and assumes that readers know the latest stable version is something different

**Pull request merged**